vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/PlotOptionsSankey.java

```diff
@@ -0,0 +1,767 @@
+/**
+ * Copyright 2000-2023 Vaadin Ltd.
+ *
+ * This program is available under Vaadin Commercial License and Service Terms.
+ *
+ * See <https://vaadin.com/commercial-license-and-service-terms> for the full
+ * license.
+ */
+package com.vaadin.flow.component.charts.model;
+
+import com.vaadin.flow.component.charts.model.style.Color;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.IntFunction;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Plot options for {@link ChartType#SANKEY} charts.
+ */
+public class PlotOptionsSankey extends AbstractPlotOptions {
+
+    private Color borderColor;
+    private Number borderRadius;
+    private Number borderWidth;
+    private Boolean centerInCategory;
+    private String className;
+    private Boolean clip;
+    private Color color;
+    private Boolean colorByPoint;
+    private Number colorIndex;
+    private List<Color> colors;
+    private String cursor;
+    private Number curveFactor;
+    private DashStyle dashStyle;
+    private DataLabels dataLabels;
+    private String description;
+    private Boolean enableMouseTracking;
+    private Boolean getExtremesFromAll;
+    private Label label;
+    private List<Level> levels;
+    private String linkedTo;
+    private Number linkOpacity;
+    private Number minLinkWidth;
+    private Number nodePadding;
+    private Number nodeWidth;
+    private Number opacity;
+    private Boolean selected;
+    private Boolean showCheckbox;
+    private Boolean showInLegend;
+    private Boolean skipKeyboardNavigation;
+    private States states;
+    private Boolean stickyTracking;
+    private Tooltip tooltip;
+    private Number turboThreshold;
+    private Boolean visible;
```

Comment:
There are a few options that I can see from the Highcharts API that are missing here:

- allowPointSelect
- animation
- inactiveOtherPoints
- linkColorMode: from | to | gradient
- nodeAlignment: top | center | bottom

Reply:
> linkColorMode: from | to | gradient
> nodeAlignment: top | center | bottom

These two have been added in [`highcharts` 11.2.0 release](https://github.com/highcharts/highcharts/blob/6cc902cb843a20c05b6b895eaa0d82ee8dcbbb87/changelog/highcharts/11.2.0.md?plain=1#L5-L6) and we use an older version.
I'll check if other options should be added and update the PR.

Reply:
Ah, right. I was relying on this information from their API page, but I guess it's not that reliable 🫠
![image](https://github.com/vaadin/flow-components/assets/262432/ff16652a-ba00-4010-8b5e-0fe84993164e)

Reply:
Yes, I also spent some time trying to figure out why `nodeAlignment` API didn't work 😕

Reply:
Added `allowPointSelect`, `animation` and `inactiveOtherPoints`.